ddtrace/tracer/sqlcomment_test.go

```diff
@@ -0,0 +1,94 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"strconv"
+	"strings"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSQLCommentPropagator(t *testing.T) {
+	prepareSpanContextWithSpanID := func(tracer *tracer) ddtrace.SpanContext {
+		root := tracer.StartSpan("service.calling.db", WithSpanID(10)).(*span)
+		root.SetTag(ext.SamplingPriority, 2)
+		return root.Context()
+	}
+
+	testCases := []struct {
+		name               string
+		query              string
+		mode               SQLCommentInjectionMode
+		prepareSpanContext func(*tracer) ddtrace.SpanContext
```

Comment:
```suggestion
		spanCtx ddtrace.SpanContext
```
Just set it to `nil` when it's not used, and set it to `testSpanCtx()` otherwise. Would that not work?

Reply:
It would require the tracer to be initialized before the test cases which I don't like because it makes it harder to show that test cases are isolated from each other. I did make a change which I hope you like as an acceptable compromise: added a boolean to inject a span context or not and that keeps the tracer scoped to a single test case.